edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionManager.cs

```diff
@@ -68,6 +68,14 @@ public async Task AddDeviceConnection(IIdentity identity, IDeviceProxy devicePro
             Preconditions.CheckNotNull(deviceProxy, nameof(deviceProxy));
             ConnectedDevice device = this.GetOrCreateConnectedDevice(identity);
             Option<DeviceConnection> currentDeviceConnection = device.AddDeviceConnection(deviceProxy);
+            currentDeviceConnection.ForEach(async c =>
```

Comment:
I think this won't work if edgeHub restarts because device connection is not stored.

Reply:
If edgeHub restarts, I believe we will resubscribe anyway through a different path, so we won't need to do this explicitly

Reply:
We see this does work on edgeHub restarts. I added a log statement in SessionStatePersistenceProvider where we see the subscription getting kicked off after an EH restart